infrastructure/azure-pipelines-dev.yml

```diff
@@ -8,38 +8,134 @@ resources:
-      - script: infrastructure/bin/deploy.sh -d -r $(System.DefaultWorkingDirectory)
-        displayName: Dependencies
-        env:
-          API_URI: /graphql
-          API_PROTECTED_URI: /admin/graphql
-          # We don't need Cypress in this environment, and so this avoid permission errors of installing it.
-          # See: https://docs.cypress.io/guides/getting-started/installing-cypress#Skipping-installation
-          CYPRESS_INSTALL_BINARY: 0
-
-      - task: ArchiveFiles@2
-        displayName: "Archive Files"
-        inputs:
-          rootFolderOrFile: $(System.DefaultWorkingDirectory)
-          includeRootFolder: false
-          archiveFile: "$(Build.ArtifactStagingDirectory)/Application_$(Build.BuildId).zip"
-
-      - task: PublishBuildArtifacts@1
-        displayName: "Publish Artifact: gcDigitalTalent"
-        inputs:
-          ArtifactName: gcDigitalTalent
+      name: "TBS Linux Self Hosted"
+    displayName: Test stage
+    dependsOn: Build
+    variables:
+      - name: npm_config_user_config
+        value: "$(System.DefaultWorkingDirectory)/.npmrc"
+      - name: dirResults
+        value: "$(System.DefaultWorkingDirectory)/apps/playwright/test-results"
+      - name: resourceGroupName
+        value: "rg_tbs_talent_2_PaaS"
+      - name: vertical
+        value: "dev"
+      - name: storageAccountName
+        value: "sttalclddevcacstdl"
+      - name: storageContainer
+        value: "playwright-results"
+    jobs:
+      - job: run_playwright_tests
+        displayName: Run Playwright tests
+        timeoutInMinutes: 10
+        pool:
+          name: "TBS Linux Self Hosted"
+        steps:
+          - checkout: self
+            clean: true
+            fetchDepth: 1
+
+          - task: DownloadBuildArtifacts@0
+            inputs:
+              artifactName: gcDigitalTalent
+              downloadPath: $(System.DefaultWorkingDirectory)/artifacts
+          - task: AzureCLI@2
+            displayName: "Update NPMRC Config"
+            inputs:
+              scriptType: bash
+              scriptLocation: inlineScript
+              azureSubscription: $(serviceConnectionName)
+              inlineScript: |
+                NPMRC_FILE="${{ variables.npm_config_user_config }}"
+                echo "Editing .npmrc: ${NPMRC_FILE}"
+                echo "registry=https://pkgs.dev.azure.com/tbs-sct/_packaging/TBS_Custom_Npm_Feed/npm/registry/
+                 always-auth=true" >> "${NPMRC_FILE}"
+          - task: npmAuthenticate@0
+            displayName: "00 - NPM Authenticate"
+            name: "NPM_Authenticate"
+            inputs:
+              workingFile: $(npm_config_user_config)
+
+          - script: echo ##vso[task.setvariable variable=NPM_CONFIG_USERCONFIG]$(npm_config_user_config)
+            displayName: "Set Custom .npmrc Location"
+
+          - script: CYPRESS_INSTALL_BINARY=/opt/cypress.zip pnpm install cypress
```

Comment:
Does pipeline actually run cypress?

Reply:
no .. I think we are good to remove it

Reply:
backing out this fails on the dependency installation on the pipeline.. looks like we need it